### ISSUE TITLE
[M] Create a program/function from week5 search engine that works with dictionary data

### ISSUE THREAD

**Issue by kanerv** [kanerv/TuukkaSaanaKanerva#38]
We planned to store our final project data in a dictionary format. Therefore, we need to edit our search engine to work with different a data format (list vs dictionary).

**Comment by TuukkaOT** [comment]
So I could not figure out how to go about vectorizing dictionary entries, so in order to check that the scraper works properly, I made a search engine that works with a list. There is a separate dictionary search engine but it doesn't work.

**Comment by TuukkaOT** [comment]
I realised that we don't need to have separate "test_query" functions for wildcard and exact searches, and they both can be implemented inside the test_query() function. So, I deleted the test_wc_query() function and put the if-elif for "choice" in test_query(). I also moved the vectorizer and matrix from the function to the beginning of the program so it is only done once. Now the search is significantly quicker.
Try it out when you have time and let me know if any bugs resulted from my changes.

**Comment by kanerv** [comment]
Great work Tuukka! I took your changes further because I wanted to get rid of code repetition. I separated the searching part of test_query into function called relevance_search(). It takes two queries as parameters because in wildcard search, the query used for searching is different from the one typed by user. Now the if loops also work correctly if the search term is not found.

I also made the wildcard search match better. Before when you searched for e.g. "movie" it did not match with "movie" itself but anything onwards like "movies" etc. I guess this was just a matter of choice, but it's easy to change back if needed.

I also did a minor change in the regex when tokenizing the data for the matrix. It did not tokenize words like _b-movies_ correctly, because it only had one character before the hyphen (so it matched words like _feel-good_, but not _a-class_). Now it should work better for our data.

I did testing whilst I worked, but let me know if you find more bugs! All the plots are not working properly, but I think that was not related to my changes. Or at least I hope so! 🐛
